pages/governance/proposal/index.governance.tsx

```diff
@@ -17,29 +17,51 @@ export default function DynamicProposal() {
   const id = Number(router.query.proposalId);
   const [proposal, setProposal] = useState<CustomProposalType>();
   const [ipfs, setIpfs] = useState<IpfsType>();
+  const [fetchMetadataError, setFetchMetadataError] = useState(false);
 
-  async function initialize() {
+  // TODO: Abstract out this recursive try/fallback approach so it may be used in other places where we fetch the proposal metadata and may have errors from the initial gateway
+  async function initialize(_ipfsGateway: string, _useFallback: boolean) {
```

Comment:
You could potentially get rid of the `_useFallback` param, and just compare the `_ipfsGateway` string with the props of `governanceConfig`.  So something like
```
if (_ipfsGateway === governanceConfig.ipfsGateway) {
  // try with fallback
} elseIf (_ipfsGateway === governanceConfig.fallbackIpfsGateway) {
  ...
}
```

It might read a little easier that way, I was kind of confused at first glance because when `_useFallback` is true, I think it actually means that the fallback _is_ being used, so don't try again.